### PR TITLE
Add AbortGameplay() to fix state transition race conditions during multiplayer load

### LINKED ISSUE #14181: InvalidStateChangeException: Cant change from Idle to Loaded
**Describe the bug:** 
Got this error message while starting to play in mp. I got reverted to back to the Lobbymenu while the others started playing.

**Screenshots or videos showing encountered issue:** 
image
**osu!lazer version:** 
2021.809.0
**Logs:**
[network.log](https://github.com/ppy/osu/files/6953815/network.log)
[performance.log](https://github.com/ppy/osu/files/6953816/performance.log)
[runtime.log](https://github.com/ppy/osu/files/6953817/runtime.log)
[updater.log](https://github.com/ppy/osu/files/6953818/updater.log)
[database.log](https://github.com/ppy/osu/files/6953819/database.log)

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [ ] Server deploy (containing https://github.com/ppy/osu-server-spectator/pull/97)

As discussed in / resolves https://github.com/ppy/osu/issues/14181

### CODE REVIEW

File: osu.Game/Screens/OnlinePlay/Multiplayer/Multiplayer.cs [hunk 1]
```diff
@@ -18,8 +19,24 @@ public override void OnResuming(IScreen last)
...
+            {
+                case MultiplayerUserState.Spectating:
+                    break;
+
+                case MultiplayerUserState.WaitingForLoad:
+                    client.AbortLoad();
+                    break;
+
+                default:
+                    client.ChangeState(MultiplayerUserState.Idle);
```
Comment (peppy) [inline osu.Game/Screens/OnlinePlay/Multiplayer/Multiplayer.cs:37]: I don't think we want this. As mentioned [here](https://github.com/ppy/osu/issues/14181#issuecomment-990631746), making the method `AbortGameplay` seems like a much better path forward. Having this remaining case feels like it's only a matter of time before it to becomes an issue.

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/OnlinePlay/Multiplayer/Multiplayer.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game/Screens/OnlinePlay/Multiplayer/Multiplayer.cs:37]: There's 5 possible different states:
```
Spectating
WaitingForLoad
Loaded
Playing
Results
```
Which of these would be handled via "AbortGameplay"?

Comment (peppy) [reply osu.Game/Screens/OnlinePlay/Multiplayer/Multiplayer.cs:37]: The middle three I would say.

ie. any time the user is expected to be in the current gameplay session but chooses to leave.

Comment (smoogipoo) [reply osu.Game/Screens/OnlinePlay/Multiplayer/Multiplayer.cs:39]: Imo it's a much of a muchness beyond the initial load one, but I've applied the changes. I haven't handled the `FinishedPlay` state which I forgot to mention above.

**Review by peppy — APPROVED** [review]

**Pull request merged**